File: core/src/main/java/com/satis/viewmodel/core/BaseLiveData.java
package com.satis.viewmodel.core;


import androidx.annotation.NonNull;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Observer;

public class BaseLiveData<T> extends MutableLiveData<T> {

    public int mVersion = -1;

    public void observe(@NonNull LifecycleOwner owner, @NonNull Observer<T> observer, boolean isSticky) {
        if(isSticky){
            super.observe(owner, observer);
        } else {
            super.observe(owner,new ObserverWrapper<T>(this,observer));
        }
    }

    @Override
    public void setValue(T value) {
        mVersion++;
        super.setValue(value);
    }

    @Override
    public void postValue(T value) {
        mVersion++;
        super.postValue(value);
    }
}

File: viewmodel-core/src/main/java/com/satis/viewmodel/core/BaseLiveData.kt
package com.satis.viewmodel.core

import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Observer

class BaseLiveData<T> : MutableLiveData<T>() {
    var mVersion = -1
    fun observe(owner: LifecycleOwner, observer: Observer<T>, isSticky: Boolean) {
        if (isSticky) {
            super.observe(owner, observer)
        } else {
            super.observe(owner, ObserverWrapper(this, observer))
        }
    }

    override fun setValue(value: T) {
        mVersion++
        super.setValue(value)
    }

    override fun postValue(value: T) {
        mVersion++
        super.postValue(value)
    }
}